File: app/src/main/java/com/chooloo/www/callmanager/adapter/MainPagerAdapter.java
package com.chooloo.www.callmanager.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.chooloo.www.callmanager.ui.page.PageContacts;
import com.chooloo.www.callmanager.ui.page.PageFragment;
import com.chooloo.www.callmanager.ui.page.PageRecents;

public class MainPagerAdapter extends FragmentStateAdapter {

    public MainPagerAdapter(FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @Override
    public int getItemCount() {
        return 2;
    }

    @NonNull
    @Override
    public PageFragment createFragment(int position) {
        switch (position) {
            case 0:
                return PageContacts.newInstance();
            case 1:
                return PageRecents.newInstance();
            default:
                return PageContacts.newInstance();
        }
    }

}

File: app/src/main/java/com/chooloo/www/callmanager/adapter/MainPagerAdapter.kt
package com.chooloo.www.callmanager.adapter

import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.chooloo.www.callmanager.ui.page.PageContacts
import com.chooloo.www.callmanager.ui.page.PageFragment
import com.chooloo.www.callmanager.ui.page.PageRecents

class MainPagerAdapter(fragmentActivity: FragmentActivity?) : FragmentStateAdapter(fragmentActivity!!) {
    override fun getItemCount(): Int {
        return 2
    }

    override fun createFragment(position: Int): PageFragment {
        return when (position) {
            0 -> PageContacts.newInstance()
            1 -> PageRecents.newInstance()
            else -> PageContacts.newInstance()
        }
    }
}